File: src/main/java/org/project/model/dto/ServiceDTO.java
package org.project.model.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.project.enums.operation.SortDirection;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ServiceDTO {
    private Long id;
    @NotBlank(message = "Tên dịch vụ không được để trống")
    private String productEntityName;
    @NotNull(message = "Phòng ban không được để trống")
    private Long departmentEntityId;
    private BigDecimal minPrice;
    private BigDecimal maxPrice;
    private Long minStarRating;
    private Long minReviewCount;
    private String productEntityDescription;
    private String productEntityImageUrl;
    @Positive(message = "Giá dịch vụ phải là một số dương")
    private BigDecimal productEntityPrice;
    @Valid
    private List<ProductAdditionalInfoDTO> productEntityProductAdditionalInfoEntities;
    @Valid
    private List<ServiceFeatureDTO> serviceFeatureEntities;
    private String sortFieldName;
    private SortDirection sortDirection;
}

File: src/main/resources/templates/frontend/assets/js/model/dto/ServiceDTO.js
export class ServiceDTO {
    constructor(productEntityName, minPrice, maxPrice, minStarRating, sortFieldName, sortDirection) {
        this.productEntityName = productEntityName;
        this.minPrice = minPrice;
        this.maxPrice = maxPrice;
        this.minStarRating = minStarRating;
        this.sortFieldName = sortFieldName;
        this.sortDirection = sortDirection;
    }

    static fromObject(obj) {
        const serviceDTO = new ServiceDTO();
        if ("productEntityName" in obj) {
            serviceDTO.productEntityName = obj.productEntityName;
        }
        if ("minPrice" in obj) {
            serviceDTO.minPrice = obj.minPrice;
        }
        if ("maxPrice" in obj) {
            serviceDTO.maxPrice = obj.maxPrice;
        }
        if ("minStarRating" in obj) {
            serviceDTO.minStarRating = obj.minStarRating;
        }
        if ("sortFieldName" in obj) {
            serviceDTO.sortFieldName = obj.sortFieldName;
        }
        if ("sortDirection" in obj) {
            serviceDTO.sortDirection = obj.sortDirection;
        }
        return serviceDTO;
    }
}